assembl/views/api2/discussion.py

```diff
@@ -675,13 +675,49 @@ def get_time_series_analytics(request):
     return csv_response([r._asdict() for r in results], format, fieldnames)
 
 
+@view_config(context=InstanceContext, name="extract_csv_taxonomy",
+             ctx_instance_class=Discussion, request_method='GET',
+             permission=P_DISC_STATS)
+def extract_taxonomy_csv(request):
+    extracts = db.query(models.Extract).all()
+    extract_list = []
+    filednames = ["Thematic", "Message", "Content Harvested", "Qualify by nature", "Qualify by action",
+                  "Owner of the message", "Published on", "Harvester", "Harvested on", "Nugget"]
+    for extract in extracts:
+        thematic = db.query(models.Idea).filter(models.Idea.id == extract.idea_id).first()
+        query = db.query(models.Post).filter(models.Post.id == extract.content_id).first()
+        message = query.body
+        content_harversted = extract.body
+        qualify_by_nature = extract.extract_nature
+        qualify_by_action = extract.extract_action
+        owner_of_the_message = db.query(models.User).filter(models.User.id == query.creator_id).first().body
+        published_on = str(query.creation_date)
+        harvester = db.query(models.User).filter(models.User.id == extract.owner_id).first().name
+        harvested_on = str(extract.creation_date)
+        nugget = "Yes" if extract.important else "No"
+        extract_info = {
+            "thematic": thematic,
+            "message": message,
+            "content_harvested": content_harvested,
+            "qualify_by_nature": qualify_by_nature,
+            "qualify_by_action": qualify_by_action,
+            "owner_of_the_message": owner_of_the_message,
+            "published_on": published_on,
+            "harvester": harvester,
+            "harvested_on": harvested_on,
+            "nugget": nugget,
+        }
+        extract_list.append(extract_info)
+    return csv_results(extract_list, CSV_MIMETYPE, fieldnames)
```

Comment:
have you tested the extraction ? You use `fieldnames` spelled right here so I think you have an error when you try to extract?